### ISSUE TITLE
Redirects & Captured 404s Not Recording Hits

### ISSUE THREAD

**Issue by totalfood** [aaron13100/404solution#19]
Hey Aaron, plugin is working great so far on my work site, and I've installed on my other Wordpress sites to use as well.

One small issue I noticed is that both the Page Redirects and Captured 404 URLs are not recording hits like 404 Redirect used to. I would always use that data to focus on addressing the 404s that were getting a lot of hits.

Hope you can fix this in the next update as it's very helpful. Thanks!

**Comment by aaron13100** [comment]
Hey totalfood. I'd love to help, but I'll need more specific information. For example, when I look at the "Page Redirects" or the "Captured 404 URLs" I can visit the URLs listed there and see the "Hits" count go up. Is that not what you're seeing?
thanks

**Comment by totalfood** [comment]
No, whether I sort the Hits top to bottom or vice versa, they all show up as 0 Hits. Same for the Captured 404s. See attached screenshots. I just noticed this too on the screenshots, but all of them say they were never used on the Last Used column. Let me know if I need to change a setting. Thanks!

![screen shot 2017-06-07 at 12 45 35 pm](https://user-images.githubusercontent.com/22596375/26890466-99f4183c-4b7f-11e7-9bcc-c4e4d6599a70.png)
![screen shot 2017-06-07 at 12 44 44 pm](https://user-images.githubusercontent.com/22596375/26890465-99f0eea0-4b7f-11e7-9e33-84882ee149d3.png)

**Comment by aaron13100** [comment]
Ok. Would you turn on the debug log under the options, then manually test
one of the URLS, then view the debug log and send me the results? Maybe an
error message will be there.

Thanks

-Aaron

**Comment by totalfood** [comment]
Screenshot attached of the debug log. Let me know if you need something else. Thanks!
![screen shot 2017-06-08 at 9 42 53 am](https://user-images.githubusercontent.com/22596375/26931601-0aba1152-4c2f-11e7-9185-eb3378bc1301.png)

**Comment by aaron13100** [comment]
Hey totalfood. I guess this must be the same "Illegal mix of collations" issue that scidave was having. I put in what I think might fix it in version 1.7.4. If it doesn't fix it there should now be an error message in the log when the redirects or captured URLs pages are displayed. Let me know how it goes.
thanks!

**Comment by totalfood** [comment]
Hey Aaron, I updated to 1.7.4 after we spoke. Still not logging hits. Ran debug again, hit one Page Redirect and one Captured 404, screenshot attached. 

Of note, currently I have number of redirects in the plugin: "All (5798) | Manual Redirects (3021) | Automatic Redirects (2777) | Trash (0)"... is it possible the plugin is maxed on memory?

Let me know what you think. Thanks again for your help. 

![screen shot 2017-06-12 at 9 08 10 am](https://user-images.githubusercontent.com/22596375/27035379-1acf661e-4f4f-11e7-8c08-39c6af3ea0d1.png)

**Comment by aaron13100** [comment]
hey totalfood. Does the "Logs" tab work correctly? If you view the Logs tab for the URL you tested do the recent hits appear there?
thanks

**Comment by aaron13100** [comment]
Are you able to offer access to your site or a duplicate site for me to test with?
thanks

**Comment by totalfood** [comment]
Logs screenshot attached... we're the 173 IP. Let me know if you need any other screenshots to help diagnose.

For your reference, we do have the Redirect option of Yoast SEO Premium activated, but it does not do auto 301 like your plugin. It only does it when we change a post title, or the Regex Redirect which seems to have mixed results. 

Thanks! Mark

**Comment by totalfood** [comment]
![screen shot 2017-06-12 at 1 21 27 pm](https://user-images.githubusercontent.com/22596375/27046557-e9f8b3fa-4f72-11e7-81bd-2dbd13bc6407.png)

**Comment by aaron13100** [comment]
Hey Mark. I'm not so much interested in the logs as I am interested in whether selecting a specific URL with the "View Logs For:" options on the Logs page works or not. On that page does selecting a specific URL and clicking "View Logs" show only the logs for that URL afterwards? If the specific URL you choose is the one you tested, can you see the most recent request appear there?

Also, are you able to offer access to your site or a duplicate site for me to test with?
thanks

**Comment by totalfood** [comment]
Understood. I’d prefer not to post a un/pw to a bulletin board, so please email me off Github and I will send you a login. Thanks!

**Comment by aaron13100** [comment]
Mark,

There's good news and there's bad news. The good news is that the problem seems to be resolved with version 1.8.1. I can see that the issue used to be there, but it seems to have been resolved around the time you upgraded to version 1.8.1. So new requests and redirects are now logged appropriately and the "hits" will appear.

The bad news is that there was no data in the log files (for which URL a user had requested) up to the exact time of the upgrade to version 1.8.1. So unfortunately, that history has been lost. I removed the lines from the database with the missing data.

On the other hand it's a clean slate or a fresh start as far your log goes. Maybe that's good. 

Also I noticed another issue with the daily cleanup not running thanks to poking around today. So I'll be able to fix that before anyone else notices, perhaps :)  

So it looks like it should work as of now. If there are any issues now then the data must be magically disappearing somehow. Let me know if that happens.

thanks
